api/datetimeginhandler/datetime.go

```diff
@@ -0,0 +1,37 @@
+package datetime
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+func SetupRouter() *gin.Engine {
+
+	r := gin.Default()
+	r.HandleMethodNotAllowed = true
+	r.GET("/datetime", GetDate)
+
+	return r
+}
+
+// GetDateAndTime takes gin context
+// and respond the current date and time in the format
+// "2006-01-02 15:04:05" to the response writer.
+// If the request is valid, it responds with a 200 OK status and the current date and time.
+// If the request method is not GET, it responds with a 405 Method Not Allowed status.
+// If the request URL is not "/datetime", it responds with a 404 Not Found status.
+func GetDate(c *gin.Context) {
+	if c.Request.URL.Path != "/datetime" {
+		http.NotFound(c.Writer, c.Request)
+	}
+
+	if c.Request.Method != "GET" {
```

Comment:
you don't need to check the request's method here

Reply:
done